### PR TITLE
Apply SSH generate/create key name validation

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Apply SSH generate/create key name validation

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/14218
depends on https://github.com/eclipse-theia/theia/issues/6245

### CODE REVIEW

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts [hunk 1]
```diff
@@ -55,10 +55,16 @@ export async function start() {
...
 }
 
-const getHostName = async () => {
-    const hostName = await theia.window.showInputBox({ placeHolder: 'Please provide a Host name e.g. github.com' });
-    return hostName ? hostName : '';
-};
+const hostNamePattern = new RegExp('[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*');
+
+const getHostName = async () => await theia.window.showInputBox({
+    placeHolder: 'Please provide a Host name e.g. github.com',
```
Comment (benoitf) [inline plugins/ssh-plugin/src/ssh-plugin-backend.ts]:
```suggestion
    placeHolder: 'Please provide a hostname e.g. github.com',
```

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts [hunk 2]
```diff
@@ -55,10 +55,16 @@ export async function start() {
...
+    validateInput: text => {
+        if (!hostNamePattern.test(text)) {
+            return 'Not a valid Host name';
```
Comment (benoitf) [inline plugins/ssh-plugin/src/ssh-plugin-backend.ts]:
```suggestion
            return 'Invalid hostname';
```

**Comment by benoitf** [comment]
FYI the build is failing (and happy path as well)

**Comment by vinokurig** [comment]
crw-ci-test

**Pull request merged**